python/meep.i

```diff
@@ -1028,6 +1028,16 @@ void _get_gradient(PyObject *grad, PyObject *fields_a, PyObject *fields_f, PyObj
     $1 = (std::complex<double> *)array_data($input);
 }
 
+// typemaps for gaussianbeam
+
+%typecheck(SWIG_TYPECHECK_POINTER, fragment="NumPy_Fragments") std::complex<double> E0[3] {
+    $1 = is_array($input);
+}
+
+%typemap(in) std::complex<double> E0[3] {
+    std::complex<double> *pyE0  = (std::complex<double> *)array_data($input);
+    $1[0] = pyE0[0]; $1[1] = pyE0[1]; $1[2] = pyE0[2];
```

Comment:
If you are going with the numpy array approach, change to:

```
$1 = (std::complex<double> *)array_data($input);
```

This is likely the source of the segfault.

Reply:
Yes, this fixed it.